packages/teams-js/src/public/mail/handoff.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Used to interact with mail capability, including opening and composing mail.
+ * @module
+ *
+ * @beta
+ */
+
+import { callFunctionInHost } from '../../internal/communication';
+import { validateEmailAddress } from '../../internal/emailAddressValidation';
+import { ensureInitialized } from '../../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../../internal/telemetry';
+import { FrameContexts } from '../constants';
+import { runtime } from '../runtime';
+import { ISerializable } from '../serializable.interface';
+import { ComposeMailParams, ComposeMailType } from './mail';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const mailTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * Extended parameters for {@link composeMail}, including support for external handoff.
+ *
+ * This interface wraps {@link ComposeMailParamsWithHandoff} to provide additional functionality for scenarios
+ * where an external handoff is needed, such as transferring a draft email created in BizChat.
+ *
+ * @see {@link ComposeNewParams} for parameters when composing a new mail item.
+ * @see {@link ComposeReplyOrForwardParams} for reply or forward-specific parameters.
+ * @see {@link ComposeMailType} for supported mail operation types.
+ *
+ * @beta
+ */
+export interface ComposeMailParamsWithHandoff {
+  /**
+   * Base parameters for composing a mail item.
+   */
+  composeMailParams: ComposeMailParams;
+  /**
+   * Use this endpoint to retrieve the handoff payload when BizChat creates an email draft for external handoff.
+   */
+  handoffId: string;
+}
+
+/**
+ * Validates an array of email addresses.
+ * For ComposeMailType.New, it is valid to pass empty arrays for toRecipients, ccRecipients, and bccRecipients.
+ * This will result in a new email with handed-off (pre-populated) body content but no pre-populated recipients.
+ *
+ * @param emails - An optional array of email addresses to validate.
+ * @throws Error with a message describing if the email address is invalid.
+ *
+ * @beta
+ */
+function validateEmails(emails?: string[]): boolean {
```

Comment:
this is very minor but I don't think this function needs to return a boolean. If it had a return type of void that would probably be more correct - it would return nothing, but if any email addresses are invalid it throws an error. Alternatively, you could have it return a boolean and then catch errors in here and return false if one occurs. I'd do option 1, but it's up to you